src/core/announcements/validation.ts

```diff
@@ -0,0 +1,206 @@
+ *
+ * @param obj - The object to check
+ * @returns True if the object is a ProfileAnnouncement, otherwise false
+ */
+const isProfileAnnouncement = (obj: unknown): obj is ProfileAnnouncement => {
+  if (!isRecord(obj)) return false;
+  if (obj["announcementType"] != AnnouncementType.Profile) return false;
+  if (!isDSNPUserId(obj["fromId"])) return false;
+  if (!isString(obj["url"])) return false;
+  if (!isString(obj["contentHash"])) return false;
+
+  return true;
+};
+
+/**
+ * isAnnouncement() is a type check for Announcement
+ *
+ * @param obj - The object to check
+ * @returns True if the object is a Announcement, otherwise false
+ */
+const isAnnouncement = (obj: unknown): obj is Announcement => {
+  if (!isRecord(obj)) return false;
+  if (!isAnnouncementType(obj["announcementType"])) return false;
+
+  const validators = {
+    [AnnouncementType.GraphChange]: isGraphChangeAnnouncement,
+    [AnnouncementType.Broadcast]: isBroadcastAnnouncement,
+    [AnnouncementType.Reply]: isReplyAnnouncement,
+    [AnnouncementType.Reaction]: isReactionAnnouncement,
+    [AnnouncementType.Profile]: isProfileAnnouncement,
+  };
+
+  return validators[obj["announcementType"]](obj);
+};
+
+/**
+ * isSignedAnnouncement() is a type check for SignedAnnouncements
+ *
+ * @param obj - The object to check
+ * @returns True if the object is a SignedAnnouncement, otherwise false
+ */
+export const isSignedAnnouncement = (obj: unknown): obj is SignedAnnouncement => {
+  if (!isRecord(obj)) return false;
+  if (!isString(obj["signature"])) return false;
+
+  return isAnnouncement(obj);
+};
+
+/**
+ * isValidAnnouncement() validates an announcement and its associated content.
+ * Note that this method only validates an announcement's signature, format and
+ * linked activity content. It **does not** validate any audio, image or video
+ * files linked in the activity content object, so these files should be
+ * validated separately.
+ *
+ * @param obj - A signed announcement to validate
+ * @param opts - Optional. Configuration overrides, such as from address, if any
+ * @returns True if the announcement is valid, otherwise false
+ */
+export const isValidAnnouncement = async (obj: unknown, opts?: ConfigOpts): Promise<boolean> => {
+  const fetch = requireGetFetchFunc(opts);
+  if (!isSignedAnnouncement(obj)) return false;
+
+  if (
+    !(await isSignatureAuthorizedTo(
+      obj.signature,
+      convertSignedAnnouncementToAnnouncement(obj),
+      obj.fromId,
+      Permission.ANNOUNCE,
+      undefined,
+      opts
+    ))
+  )
+    return false;
+
+  if (isBroadcastAnnouncement(obj) || isReplyAnnouncement(obj) || isProfileAnnouncement(obj)) {
+    const content = await fetch(obj["url"]).then((res) => res.text());
+    if (keccak256(content) != obj["contentHash"]) return false;
+
+    const activityContent = JSON.parse(content);
+    if (!isValidActivityContent(activityContent)) return false;
+  }
```

Comment:
Along the questions about fetch, I think this should be considered a separate piece (or at least `JSON.parse` and `isValidActivityContent`). Either that or perhaps `isValidAnnouncement` shouldn't be in core? This does feel like a composition method.

Reply:
removed